replication-api/src/main/java/com/connexta/replication/queue/CompoundQueue.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * A compound queue is an artifact created around multiple {@link SiteQueue}s allowing a worker to
+ * retrieve tasks from any of the queues based on task priorities.
+ */
+public interface CompoundQueue extends Queue {
```

Comment:
I don't think we should spend much churn over this, but this may be able to just be an implementation that is returned via the broker and a client doesn't care which queues are part of the queue. If a client needs the SiteQueue to offer/put to, then it can go individually through the broker?

Reply:
I don't think there will be any churn on this since this composite queue will not to keep track of the set of `SiteQueue`s anyway. So this is a strait and simple getter.